pilot/proxy/envoy/mixer.go

```diff
@@ -201,42 +207,48 @@ func mixerHTTPRouteConfig(role proxy.Node, instances []*model.ServiceInstance, c
 					},
 				},
 			},
+			DisableCheckCalls:  outboundRoute || mesh.DisablePolicyChecks,
+			DisableReportCalls: outboundRoute,
 		}
 
-		apiSpecs := config.HTTPAPISpecByDestination(instance)
-		model.SortHTTPAPISpec(apiSpecs)
-		for _, config := range apiSpecs {
-			sc.HttpApiSpec = append(sc.HttpApiSpec, config.Spec.(*mccpb.HTTPAPISpec))
-		}
+		// omit API, Quota, and Auth portion of service config when
+		// check and report are disabled.
+		if !sc.DisableCheckCalls || !sc.DisableReportCalls {
```

Comment:
We should not omit API/Quota/Auth when `Report` is disabled.
They are orthogonal.

Reply:
We only omit when neither `Check` or `Report` is enabled

Reply:
`sc.DisableReport` is orthogonal though. It should be only based on if `Check` is enabled.

For example, if Check if disabled, but report is enabled, we *do* want to omit.